components/eam/tools/topo_tool/cube_to_target/cube_to_target.F90

```diff
@@ -841,17 +882,70 @@ program convterr
   sgh30_target = SQRT(sgh30_target)
   WRITE(*,*) "min/max of sgh_target     : ",MINVAL(sgh_target),MAXVAL(sgh_target)
   WRITE(*,*) "min/max of sgh30_target   : ",MINVAL(sgh30_target),MAXVAL(sgh30_target)
+
+  !-----------------------------------------------------------------------------
+  ! Orographic shape parameters
+  if (calc_orographic_shape_params) then
+
+    nvar_dirOA=2+1
+    nvar_dirOL=180
+
+    allocate(oa_target(ntarget,nvar_dirOA),     stat=alloc_error)
+    allocate(oc_target(ntarget),                stat=alloc_error)
+    allocate(ol_target(ntarget,nvar_dirOL),     stat=alloc_error)
+    allocate(indexb(ntarget),                   stat=alloc_error)
+    allocate(terrout(4,ntarget,maxval(indexb)), stat=alloc_error)
+    allocate(dxy(ntarget,nvar_dirOL),           stat=alloc_error)
+    oc_target = 0
+    ol_target = 0
+    indexb    = 0
+
+    ! Orographic asymmetry
+    print*,"calculating orographic asymmetry..."
+    call orographic_asymmetry_xie2020(terr, ntarget, ncube, n, nvar_dirOA, jall, &
+                                      weights_lgr_index_all, weights_eul_index_all(:,1), &
+                                      weights_eul_index_all(:,2), weights_eul_index_all(:,3), &
+                                      weights_all, target_center_lon, target_center_lat, &
+                                      lon_terr, lat_terr, area_target, oa_target)
+    
+    ! Orographic convexity
+    print*,"calculating orographic convexity..."
+    call orographic_convexity_kim2005(terr, ntarget, ncube, n, jall, &
+                                      weights_lgr_index_all, weights_eul_index_all(:,1), &
+                                      weights_eul_index_all(:,2), weights_eul_index_all(:,3), &
+                                      weights_all, area_target, sgh_target, terr_target, oc_target)
+
+    ! Orographic effective length
+    print*,"calculating orographic effective length..."
+    do count=1,jall
+      i = weights_lgr_index_all(count)
+      indexb(i) = indexb(i)+1
+    enddo
+    call orographic_efflength_xie2020(terr, ntarget, ncube, n, jall, nlon, nlat, maxval(indexb), &
+                                      nvar_dirOL, weights_lgr_index_all, weights_eul_index_all(:,1), &
+                                      weights_eul_index_all(:,2), weights_eul_index_all(:,3), &
+                                      weights_all, target_center_lon, target_center_lat, &
+                                      target_corner_lon_deg, target_corner_lat_deg, &
```

Comment:
I might be missing it, but it looks like `target_corner_lon_deg` and `target_corner_lat_deg` are being used uninitialized here. That said, see above comment about whether or not it even makes sense to define an extra set of big arrays here to hold this information.

Reply:
Yea, these definitely aren't needed, so I replaced them with the variables without the `_deg` suffix

Reply:
done.